homeassistant/components/synology_dsm/__init__.py

```diff
@@ -163,6 +163,16 @@ def _async_migrator(entity_entry: entity_registry.RegistryEntry):
 
     await entity_registry.async_migrate_entries(hass, entry.entry_id, _async_migrator)
 
+    # Migrate existing entry configuration
+    if entry.data.get(CONF_VERIFY_SSL) is None:
+        _LOGGER.debug(
+            "async_setup_entry - migrate existing entry configuration for entry: %s",
+            entry.as_dict(),
+        )
```

Comment:
I don't think we need a logger here.

It will be used only one time + it will expose sensitive data.

Reply:
> Fine !
> 
> I worked on something else but I think your proposal is better.
> 
> My solution: [hacf-fr@9d84ec8](https://github.com/hacf-fr/home-assistant-core/commit/9d84ec83bf36e13d709b10c733b7c65efbaf256d)
> 
> What do you think ?

In the very beginning I still had the same idea, but I guess this approach will never change the configuration file it self and therefore I searched for a possibility to also update the configuration file 🤔